sys/targets/targets.go

```diff
@@ -441,6 +442,14 @@ func init() {
 			if host := List[goos][runtime.GOARCH]; host != nil {
 				target.CCompiler = host.CCompiler
 				target.CPP = host.CPP
+				if goos == "freebsd" {
+					// For some configurations -no-pie is passed to the compiler,
+					// which is not used by clang.
+					// Ensure clang does not complain about it.
+					target.CrossCFlags = append(target.CrossCFlags, "-Wno-unused-command-line-argument")
+					// clang complains about remove_dir() in executor/common.h. Silence this.
```

Comment:
This warning is very intentional. It's supposed to prevent lots of unused stuff in C reproducers. The minimal reproducer may contain just main() function, maximal -- dozens of functions. Functions are excluded based of these ifdef's:

```
#if GOOS_akaros || GOOS_netbsd || GOOS_freebsd || GOOS_openbsd || GOOS_test
#if SYZ_EXECUTOR || SYZ_EXECUTOR_USES_FORK_SERVER && SYZ_REPEAT && SYZ_USE_TMP_DIR
....
static void remove_dir(const char* dir)
```

There must be something wrong with the ifdefs. There is some combination where we include remove_dir, but don't use it. We should fix the ifdef's instead.

Reply:
I consider this a bad hack, but I wanted to have a starting point for discussion. There are two build issues which need to be addressed:

1. When running `gmake test`, some test programs are compiled with `-no-pie` coming from https://github.com/google/syzkaller/blob/master/sys/targets/targets.go#L86 and https://github.com/google/syzkaller/blob/master/sys/targets/targets.go#L100. When using `clang`, it warns the it is ignoring some unused command line arguments. A real fix for me would only add `no-pie` when compiling on Linux with ASan and ASLR on ppc64le. But I neither know how to achieve this nor I have a test machine (I do have a Power9 machine, but it runs FreeBSD).

2. I'll look into the `remove_dir()` issue as you described. I'll bring that up as a separate PR.

Reply:
Re -no-pie, short term I would go with your current solution.
Longer term, @ramosian-glider is working on adding clang support for linux in #1732. I assume it may affect linux as well (?). The option discussed is to split these flags into common/gcc/clang:
https://github.com/google/syzkaller/pull/1732#issuecomment-627507995
Then I assume this may be resolved in a better way.

Reply:
I agree.

Reply:
OK,  the `remove_dir()` issues is resolved in https://github.com/google/syzkaller/commit/c7875f08e1736a5b3e9f615c8821ac86f3153350 and https://github.com/google/syzkaller/commit/b3fe7470a482f2a877cbe43f46636153031b20e3. Sorry for needing two commit, no idea why I didn't catch the typo.

Reply:
We have https://github.com/google/syzkaller/blob/master/pkg/csource/csource_test.go#L181 which have been caught it... except that it did not caught that exact misspelling :) And we can't bark on "SYS_" because it's used for some syscalls on linux...

Reply:
That way I found it, unfortunately after the first PR had been merged. It took me two or three hours to find it. I just did not see it.